### ISSUE TITLE
MySQLiConnector: mysqli_stmt::get_result is undefined if not using mysqlnd driver

### ISSUE THREAD

**Issue by halkyon** [silverstripe/silverstripe-framework#3426]
```
Fatal error: Call to undefined method mysqli_stmt::get_result from context MySQLiConnector in /var/www/ss-master/framework/model/connect/MySQLiConnector.php on line 248
```

This doesn't work on some environments, specifically those that use libmysql instead of mysqlnd (e.g. HHVM). More info on that here: http://php.net/manual/en/mysqli-stmt.get-result.php.

**Comment by Martimiz** [comment]
Using MAMP (PRO) with mysqlnd also doesn't seem to be an option - unless you compile your own distribs for it, which may not be everyone's cup of tea (not mine anyway :( )

**Comment by flashbackzoo** [comment]
I ran into this yesterday using Ubuntu 14.04. A quick(ish) workaround is to install and use SQLite3.

```
sudo apt-get update
sudo apt-get install sqlite3 libsqlite3-dev
sudo apt-get install php5-sqlite3
sudo service apache2 restart
````

Then include the [silverstripe-sqlite3](https://github.com/silverstripe-labs/silverstripe-sqlite3) module.

One gotcha I ran into was database permissions. The Apache user (`www-data` on Ubuntu) has to have write permissions to the database AND the database's parent directory. So watch that.

**Comment by tractorcow** [comment]
I'm working on a solution to this issue, although I'm also looking at the performance costs of using unbuffered statements for all queries. I need to spend some time to test my solutions. ;D

**Comment by tractorcow** [comment]
Possible solution at https://github.com/silverstripe/silverstripe-framework/pull/4143

It took ages to get all the segfaults out of this!

**Comment by sminnee** [comment]
I've merged #4154 that fixes this.

**Comment by JohnMaina** [comment]
I have noted that many web hosts using Linux servers don't have mysqlnd driver. It comes as a shock for a developer using windows to find that he will need to re do the whole app using something like bind_results()......Why is it too hard for the hosts to install mysqlnd on their servers?
